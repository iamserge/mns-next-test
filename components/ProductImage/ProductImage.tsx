import { useState } from "react";
import ProductBadge from "../ProductBadge/ProductBadge";
const ProductImage = ({image_key, name, offers}) => {
    const imgUrl = `https://asset1.cxnmarksandspencer.com/is/image/mands/${image_key}`;
    const [loaded, setLoaded ] = useState(false);
    return (
        <div className={"relative max-h-10 image-container" + (loaded ? " p-0" : " loading-image-container")}>
            <img loading="lazy" className="w-full min-h-full" src={imgUrl} alt={name} onLoad={()=>{setLoaded(true)}} />
            <ProductBadge productOffers={offers} />     
        </div>
    )
}

export default ProductImage;
