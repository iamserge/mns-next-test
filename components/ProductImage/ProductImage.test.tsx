import renderer from "react-test-renderer";
import ProductImage from "./ProductImage";

describe("ProductImage", () => {
  it("renders correctly", () => {
    const component = renderer.create(<ProductImage offers={["1", "2"]} name="test" image_key="123" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
