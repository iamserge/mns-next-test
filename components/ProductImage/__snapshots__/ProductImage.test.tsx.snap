// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage renders correctly 1`] = `
<div
  className="relative max-h-10 image-container loading-image-container"
>
  <img
    alt="test"
    className="w-full min-h-full"
    loading="lazy"
    onLoad={[Function]}
    src="https://asset1.cxnmarksandspencer.com/is/image/mands/123"
  />
</div>
`;
